test/test_tensor.py

```diff
@@ -154,5 +155,11 @@ def test_random_fns_are_deterministic_with_seed(self):
         b = random_fn(10,10).realize()
         np.testing.assert_allclose(a.numpy(), b.numpy())
 
+  def test_zeros_like_has_same_dtype(self):
+    for dtype in dtypes.__annotations__:
+      a = Tensor([1, 2, 3], dtype=dtypes.float16)
+      b = Tensor.zeros_like(a)
+      self.assertTrue(a.dtype == b.dtype)
```

Comment:
remove loop, don't really need to test every dtype (and currently you have a bug, `dtype` var is unused)

just have 2 cases and add shape assert